data.php

```diff
@@ -183,10 +185,25 @@ function setShowBlockedPermitted()
     }
 
     function getAllQueries($orderBy) {
-        global $log,$gravity,$showBlocked,$showPermitted;
+        global $log,$gravity,$showBlocked,$showPermitted,$whitelist,$blacklist;
         $allQueries = array("data" => array());
         $dns_queries = getDnsQueriesAll($log);
-        $gravity_domains = getGravityDomains($gravity);
+        $gravity_domains = getDomains($gravity);
+        $whitelist_domains = getDomains($whitelist);
+        $blacklist_domains = getDomains($blacklist);
+
+        // Remove whitelisted domains
+        foreach($gravity_domains as $domain) {
```

Comment:
Looping over all `$gravity_domains` is looping over hundreds of thousands of entries. We should loop over `$whitelist_domains`, instead.